ros_buildfarm/ci_job.py

```diff
@@ -303,7 +303,8 @@ def _get_ci_job_config(
 
         # only Ubuntu Focal has a new enough pytest version which generates
         # JUnit compliant result files
-        'xunit_publisher_types': get_xunit_publisher_types_and_patterns(),
+        'xunit_publisher_types': get_xunit_publisher_types_and_patterns(
+            ros_version, os_name == 'ubuntu' and os_code_name != 'bionic'),
```

Comment:
to be junit compliant, does it depends on a given platform or depends on a certain pytest version?

Reply:
As the comment above this line states:

> only Ubuntu Focal has a new enough pytest version which generates JUnit compliant result files